kv/migration/all/0016_add-annotations-notebooks-to-oper-token.go

```diff
@@ -0,0 +1,147 @@
+package all
+
+import (
+	"context"
+	"encoding/json"
+	"sort"
+
+	"github.com/influxdata/influxdb/v2"
+	"github.com/influxdata/influxdb/v2/kv"
+)
+
+var Migration0016_AddAnnotationsNotebooksToOperToken = UpOnlyMigration(
+	"add annotations and notebooks resource types to operator token",
+	func(ctx context.Context, store kv.SchemaStore) error {
+		authBucket := []byte("authorizationsv1")
+
+		oprPerms := influxdb.OperPermissions()
+
+		// Find the operator token that needs updated
+
+		// There will usually be 1 operator token. If somebody has deleted their
+		// operator token, we don't necessarily want to make it so that influxdb
+		// won't start, so store a list of the found operator tokens so that it can
+		// be iterated on later.
+		opTokens := []influxdb.Authorization{}
+		if err := store.View(ctx, func(tx kv.Tx) error {
+			bkt, err := tx.Bucket(authBucket)
+			if err != nil {
+				return err
+			}
+
+			cursor, err := bkt.ForwardCursor(nil)
+			if err != nil {
+				return err
+			}
+
+			return kv.WalkCursor(ctx, cursor, func(_, v []byte) (bool, error) {
+				var t influxdb.Authorization
+				if err := json.Unmarshal(v, &t); err != nil {
+					return false, err
+				}
+
+				// If the permissions list of this token + the extra permissions an
+				// operator token for annotations and notebooks would have matches the
+				// full list of operator permissions for a 2.1 operator token, we must
+				// be dealing with a 2.0.x operator token, so add it to the list.
+				if permListsMatch(oprPerms, append(t.Permissions, extraPerms()...)) {
```

Comment:
Annoying as it is, I think it'd be better to hard-code the expected final list here. Assuming we add more resource-types in versions after 2.1, the contents of `oprPerms` will continue to grow as defined here. Once that happens, this conditional will no longer match for any users upgrading from `2.0`->`2.N`, so the notebooks & annotations resources won't get added to their tokens.

Reply:
Good call, hadn't thought about the case where there'd be more than one generate of additional resource types to migrate into. Made that change!